File: 36.py
# -*- coding:utf-8 -*-
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None
class Solution:
    def __init__(self):
        self.pEnd = None
        self.Head = None
    def core(self,pNode:TreeNode):
        self.core(pNode.left)
        if(self.Head==None):
            self.pEnd = pNode
            self.Head = pNode
        else:
            self.pEnd.right = pNode
            pNode.left = self.pEnd
            self.pEnd = pNode
        self.core(pNode.right)

    def Convert(self, pRootOfTree):
        self.core(pRootOfTree)
        return self.Head

File: 36.cpp
//
// Created by zb198 on 2019/9/22.
//

struct TreeNode {
	int val;
	struct TreeNode *left;
	struct TreeNode *right;
	TreeNode(int x) :
			val(x), left(nullptr), right(nullptr) {
	}
};
class Solution {
public:
    TreeNode* pEnd= nullptr;
    TreeNode* Head = nullptr;
    void core(TreeNode* pNode){
        if(pNode== nullptr)
            return;
        core(pNode->left);
        if(Head== nullptr){
            Head = pNode;
            pEnd = pNode;
        } else{
            pEnd->right=pNode;
            pNode->left=pEnd;
            pEnd = pNode;
        }
        core(pNode->right);
    }
    TreeNode* Convert(TreeNode* pRootOfTree)
    {
        core(pRootOfTree);
        return Head;
    }
};

